skins/oasis/modules/OasisController.class.php

```diff
@@ -282,6 +339,28 @@ private function loadJs() {
 			$this->globalBlockingScripts .= "<script type=\"$wgJsMimeType\" src=\"$blockingFile\"></script>";
 		}
 
+		// move JS files added to OutputPage to list of files to be loaded
+		$scripts = RequestContext::getMain()->getSkin()->getScripts();
+
+			foreach ( $scripts as $s ) {
+			//add inline scripts to jsFiles and move non-inline to the queue
+			if ( !empty( $s['url'] ) ) {
+				// FIXME: quick hack to load MW core JavaScript at the top of the page - really, please fix me!
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1134)